### PR TITLE
[release-0.49] [Bug-fix]: Windows VM HyperV Reenlightenment enabled fails to migrate

### PR DESCRIPTION
**Pull request by iholder101**
**What this PR does / why we need it**:
Manual backport of this PR: https://github.com/kubevirt/kubevirt/pull/8201.

**Release note**:

```release-note
[Bug-fix]: Windows VM with WSL2 guest fails to migrate
```

### CODE REVIEW

**Comment by iholder101** [comment]
/cc @enp0s3 
Can you PTAL?

**Review by enp0s3 — APPROVED** [review]
/approve

**Comment by acardace** [comment]
/hold
I think this will make re-enlightenment VMIs not migratable when upgrading to the latest 0.49, @iholder101 can you confirm?

**Comment by acardace** [comment]
@iholder101 maybe we should add topology hints for existing VMIs as well so that we don't break live migration on upgrades, wdyt?

**Comment by iholder101** [comment]
> I think this will make re-enlightenment VMIs not migratable when upgrading to the latest 0.49, @iholder101 can you confirm?

That's correct.
More accurately, after upgrading to latest 0.49 such VMIs would be able to migrate only to nodes supporting the lowest tsc of the cluster.

I'll just note that even in current 0.49 although migrating such VMs is possible - it is dangerous and discouraged by QEMU. That's the reason they broke backward compatibility. So while it will limit the users, it would also protect him from bugs w.r.t. clock being consistent after migrations.

> maybe we should add topology hints for existing VMIs as well so that we don't break live migration on upgrades, wdyt?

As we discussed offline, you're going to this logic in a separate PR. Great stuff :+1: 

> /hold

Let's keep this PR holded for a while to both let others have a chance to look and in order to wait for your follow-up PR to kick in.

**Comment by acardace** [comment]
@iholder101 this PR together with https://github.com/kubevirt/kubevirt/pull/9201 should fix the upgrade issue, PTAL.

**Comment by iholder101** [comment]
/hold
CI is continuously failing. taking a look.

**Comment by enp0s3** [comment]
@iholder101 Hi, I see that VMI name is `test-vmi` perhaps this name was re-used, which can lead to that

**Comment by iholder101** [comment]
> @iholder101 Hi, I see that VMI name is `test-vmi` perhaps this name was re-used, which can lead to that

Thanks @enp0s3!
Let's wait for your fix PR and merge this afterwards.

**Comment by iholder101** [comment]
> /test pull-kubevirt-e2e-k8s-1.22-sig-compute-migrations

Hey @acardace!
The issues we see here are fixed in @enp0s3's PR: https://github.com/kubevirt/kubevirt/pull/9222.
I suggest to wait for it to merge then rerun the tests

**Comment by iholder101** [comment]
https://github.com/kubevirt/kubevirt/pull/9222 is now merged.
/unhold
/retest

File: tests/migration_test.go [hunk 1]
```diff
@@ -3651,6 +3652,46 @@ var _ = Describe("[Serial][rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][leve
...
+				EventuallyWithOffset(1, func() bool {
+					vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					return topology.AreTSCFrequencyTopologyHintsDefined(vmi)
+				}, 90*time.Second, 3*time.Second).Should(BeTrue(), fmt.Sprintf("tsc frequency topology hints are expected to exist for vmi %s", vmi.Name))
+			}
+
+			It("invtsc feature exists", func() {
+				vmi := libvmi.New("test-vmi",
```
Comment (enp0s3) [inline tests/migration_test.go:3670]: @iholder101 As #9222 got merged, you don't need to specify the VMI name in `libvmi.New("test-vmi",` anymore

File: tests/migration_test.go [hunk 2]
```diff
@@ -3651,6 +3652,46 @@ var _ = Describe("[Serial][rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][leve
...
+				vmi.Spec.Domain.CPU = &v1.CPU{Features: []v1.CPUFeature{
+					{Name: "invtsc", Policy: "require"},
+				}}
+				vmi = runVMIAndExpectLaunch(vmi, 240)
+
+				expectTopologyHintsToBeSet(vmi)
+			})
+
+			It("HyperV reenlightenment is enabled", func() {
+				vmi := libvmi.New("test-vmi")
```
Comment (enp0s3) [inline tests/migration_test.go:3682]: same here

File: tests/migration_test.go (hunk 1 above)
Comment (iholder101) [reply tests/migration_test.go:3670]: I don't understand how it even compiles :0
I thought that the bot would rebase automatically, let me rebase manually.

**Pull request merged**